packages/framework/immutable-merge/src/Merge.ts

```diff
@@ -51,76 +128,78 @@ export type IMergeRecursionHandler = (options: IMergeOptions, ...objs: any[]) =>
  * be `{}` - no recursion, `{ depth: -1 }` - recurse infinitely
  * @param objs - an array of objects to merge together
  */
-export function immutableMergeCore(options: IMergeOptions, ...objs: any[]): any {
-  const setToMerge = objs.filter(v => _isNonArrayObject(v, true));
-  const { depth = 0, recurse } = options;
-  const processSingle = options.processSingles && setToMerge.length === 1;
+function immutableMergeWorker(mergeOptions: RecursionOption | MergeOptions, singleMode: boolean, ...objs: any[]): any {
```

Comment:
You may want to add to the comment an @ param for singleMode

Reply:
Good catch, the comment was out of date.